homeassistant/components/eq3btsmart/__init__.py

```diff
@@ -28,47 +24,38 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-type Eq3ConfigEntry = ConfigEntry[Eq3ConfigEntryData]
-
-
 async def async_setup_entry(hass: HomeAssistant, entry: Eq3ConfigEntry) -> bool:
     """Handle config entry setup."""
-
-    mac_address: str | None = entry.unique_id
-
-    if TYPE_CHECKING:
-        assert mac_address is not None
-
-    eq3_config = Eq3Config(
-        mac_address=mac_address,
-    )
+    mac_address: str = entry.data.get(CONF_MAC_ADDRESS, entry.unique_id)
 
     device = bluetooth.async_ble_device_from_address(
         hass, mac_address.upper(), connectable=True
     )
 
     if device is None:
-        raise ConfigEntryNotReady(
-            f"[{eq3_config.mac_address}] Device could not be found"
+        raise ConfigEntryNotReady(f"[{mac_address}] Device could not be found")
+
+    if not entry.data:
+        hass.config_entries.async_update_entry(
+            entry,
+            data={CONF_MAC_ADDRESS: mac_address},
         )
```

Comment:
Why do we do this? I would say that this is something we should do in a migration, but I am more curious to the rationale